xskillscore/core/comparative.py

```diff
@@ -0,0 +1,168 @@
+import warnings
+
+import scipy.stats as st
+import xarray as xr
+
+from .deterministic import mae
+
+metric_larger_is_better = (
+    []
+)  # no xskillscore metric that allow dim=[] # pearson_r not allowed for sign_test see DelSole and Tippett (2016)
+metric_smaller_is_better = [
+    'mae',
+    'rmse',
+    'median_absolute_error',
+    'mse',
+    'brier_score',
+    'threshold_brier_score',
+    'crps_ensemble',
+    'median_absolute_error',
+    'smape',
+    'mape',
+    'rps',
+]
+
+
+def sign_test(
+    forecast1,
+    forecast2,
+    observation,
+    dim=None,
+    categorical=False,
+    alpha=0.05,
+    metric=mae,
+):
+    """
+        Returns the Delsole and Tippett sign test over the given time dimension.
+
+        Parameters
+        ----------
+        forecast1 : xarray.Dataset or xarray.DataArray
+            forecast1 to be compared to observation
+        forecast2 : xarray.Dataset or xarray.DataArray
+            forecast2 to be compared to observation
+        observation : xarray.Dataset or xarray.DataArray
+            observation to be compared to both forecasts
+
+            if None, then assume that forecast1 and forecast2 have already been compared
```

Comment:
Why don't we use `metric` to determine whether forecasts have already been compared to observation. That is, if `metric==None`, `observation` is not used. Then we can get rid of the optional string inputs for `observation`
```python
def sign_test(forecast1, forecast2, observation=None, dim=None, metric=mae, orientation="negative", categorical=False, alpha=0.05):
```

Reply:
I agree that ideally `orientation` and `categorical` would be known from `metric`, but I don't think that's feasible, especially since we're allowing user-specified `metric`s (which I think we want to do)

Reply:
got rid of observation str. is there a difference whether `observation=None` or `metric=None` to specify that comparison has already taken place?

Reply:
I guess not particularly, but to me `metric=None` is more intuitive, implying "no metric is needed because the metric has already been calculated". Also, currently, a user could pre-calculate, for example, the Brier scores for their forecasts, and pass them to `sign_test` with `observations=None` and `metric=<not Brier score>`. Obviously whatever is provided as `metric` in this case would not be used, but this might still cause confusion. 

What about something like:

```python 
def sign_test(forecast1, forecast2, 
              observation=None, dim=None, metric=None, 
              orientation="negative", categorical=False, alpha=0.05):

if metric:
    if observations:
        # Compare the forecasts and observations using metric
    else:
        raise ValueError("observations must be provided when metric is provided")
else
    if observations:
        raise warning("Ignoring provided observations because no metric was provided")

# Adjust for orientation

# Compute sign test
```
This way the default case is the `climpred`-type case, where the forecasts have already been compared to observations, but the user also has the option to specify both `observation` and a `metric` if they wish to run the `sign_test` on raw forecasts.